Lib/fontTools/merge.py

```diff
@@ -359,31 +359,57 @@ def merge(self, m, tables):
 @_add_method(ttLib.getTableClass('cmap'))
 def merge(self, m, tables):
 	# TODO Handle format=14.
-	cmapTables = [(t,fontIdx) for fontIdx,table in enumerate(tables) for t in table.tables if t.isUnicode()]
-	# TODO Better handle format-4 and format-12 coexisting in same font.
-	# TODO Insert both a format-4 and format-12 if needed.
-	module = ttLib.getTableModule('cmap')
-	assert all(t.format in [4, 12] for t,_ in cmapTables)
-	format = max(t.format for t,_ in cmapTables)
-	cmapTable = module.cmap_classes[format](format)
-	cmapTable.cmap = {}
-	cmapTable.platformID = 3
-	cmapTable.platEncID = max(t.platEncID for t,_ in cmapTables)
-	cmapTable.language = 0
-	cmap = cmapTable.cmap
+	# Only merges 4/3/1 and 12/3/10 subtables, ignores all other subtables
+	cmapTables = [(t,fontIdx)
+		for fontIdx,table in enumerate(tables)
+			for t in table.tables
+				if (t.format, t.platformID, t.platEncID) in [(12,3,10), (4,3,1)]]
+
+	# Build a unicode mapping, then decide which format is needed to store it.
+	cmap = {}
 	for table,fontIdx in cmapTables:
-		# TODO handle duplicates.
+		# If there is a format 12 table for the same font, ignore the format 4 table
+		if table.format == 4 and (12, fontIdx) in [(t.format, fontIdx) for t,fontIdx in cmapTables]:
+			continue
+		# handle duplicates
 		for uni,gid in table.cmap.items():
 			oldgid = cmap.get(uni, None)
 			if oldgid is None:
 				cmap[uni] = gid
 			elif oldgid != gid:
 				# Char previously mapped to oldgid, now to gid.
 				# Record, to fix up in GSUB 'locl' later.
-				assert m.duplicateGlyphsPerFont[fontIdx].get(oldgid, gid) == gid
-				m.duplicateGlyphsPerFont[fontIdx][oldgid] = gid
+				if m.duplicateGlyphsPerFont[fontIdx].get(oldgid, gid) == gid:
+					m.duplicateGlyphsPerFont[fontIdx][oldgid] = gid
+				else:
+					# char previously mapped to oldgid but already remapped to a different gid,
+					# save new gid as an alternate
+					# TODO: save the alternates in a 'aalt' table. currently just dropped.
```

Comment:
Removed. The warning output is now like
```
$ pyftmerge NotoNaskhArabic-Regular.ttf NotoSerif-Regular.ttf
WARNING: Dropped mapping from codepoint 0xa0 to glyphId 'uni00A0#1'
WARNING: Dropped mapping from codepoint 0xfeff to glyphId 'uniFEFF#1'
```

Is `aalt` not useful because it's not implemented anywhere, or am I misinterpreting the spec on it?